clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -880,7 +893,7 @@ public synchronized boolean isPaused(TopicPartition tp) {
 
     synchronized boolean isFetchable(TopicPartition tp) {
         TopicPartitionState assignedOrNull = assignedStateOrNull(tp);
-        return assignedOrNull != null && assignedOrNull.isFetchable();
+        return assignedOrNull != null && isFetchableAndSubscribed(tp, assignedOrNull);
```

Comment:
nit: This is a bit odd. If we call `isFetchableAndSubscribed`, we know that `assignedOrNull` is specifically not null. I would just rename that variable to `tps` or such, so there's no implication of nullness.